diff --git a/ports/rapidjson/portfile.cmake b/ports/rapidjson/portfile.cmake
index 96f2d6bde..3daa000cd 100644
--- a/ports/rapidjson/portfile.cmake
+++ b/ports/rapidjson/portfile.cmake
@@ -4,7 +4,6 @@ vcpkg_from_github(
     REPO Tencent/rapidjson
     REF 232389d4f1012dddec4ef84861face2d2ba85709 # accessed on 2022-06-28
     SHA512 0d7d751179abdaa6ebf6167d522651a2d13bc024d20c7e3f775c7397a8aab4cd866a6c91a55521ad7847e910822fcf982625c7308c74f5df663e6fd81336c9fc
-    FILE_DISAMBIGUATOR 2
     HEAD_REF master
 )
 
@@ -12,21 +11,21 @@ vcpkg_from_github(
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
-        -DRAPIDJSON_BUILD_DOC:BOOL=OFF
-        -DRAPIDJSON_BUILD_EXAMPLES:BOOL=OFF
-        -DRAPIDJSON_BUILD_TESTS:BOOL=OFF
-        -DCMAKE_INSTALL_DIR:STRING=cmake
+        -DRAPIDJSON_BUILD_DOC=OFF
+        -DRAPIDJSON_BUILD_EXAMPLES=OFF
+        -DRAPIDJSON_BUILD_TESTS=OFF
+        -DCMAKE_INSTALL_DIR=cmake
 )
 vcpkg_cmake_install()
 
-vcpkg_cmake_config_fixup(CONFIG_PATH cmake)
+vcpkg_cmake_config_fixup(PACKAGE_NAME RapidJSON CONFIG_PATH lib/cmake/RapidJSON)
+vcpkg_fixup_pkgconfig()
 
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/lib" "${CURRENT_PACKAGES_DIR}/debug" "${CURRENT_PACKAGES_DIR}/share/doc")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/share/doc")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include" "${CURRENT_PACKAGES_DIR}/debug/share")
 
-file(READ "${CURRENT_PACKAGES_DIR}/share/rapidjson/RapidJSONConfig.cmake" _contents)
-string(REPLACE "\${RapidJSON_SOURCE_DIR}" "\${RapidJSON_CMAKE_DIR}/../.." _contents "${_contents}")
-string(REPLACE "set( RapidJSON_SOURCE_DIR \"${SOURCE_PATH}\")" "" _contents "${_contents}")
-string(REPLACE "set( RapidJSON_DIR \"${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel\")" "" _contents "${_contents}")
-file(WRITE "${CURRENT_PACKAGES_DIR}/share/rapidjson/RapidJSONConfig.cmake" "${_contents}\nset(RAPIDJSON_INCLUDE_DIRS \"\${RapidJSON_INCLUDE_DIRS}\")\n")
+file(READ "${CURRENT_PACKAGES_DIR}/share/RapidJSON/RapidJSONConfig.cmake" _contents)
+string(REPLACE "\${RapidJSON_CMAKE_DIR}/../../../include" "\${RapidJSON_CMAKE_DIR}/../../include" _contents "${_contents}")
+file(WRITE "${CURRENT_PACKAGES_DIR}/share/RapidJSON/RapidJSONConfig.cmake" "${_contents}\nset(RAPIDJSON_INCLUDE_DIRS \"\${RapidJSON_INCLUDE_DIRS}\")\n")
 
 file(INSTALL "${SOURCE_PATH}/license.txt" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
diff --git a/ports/rapidjson/vcpkg.json b/ports/rapidjson/vcpkg.json
index ad6c7b894..c62de944b 100644
--- a/ports/rapidjson/vcpkg.json
+++ b/ports/rapidjson/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "rapidjson",
   "version-date": "2022-06-28",
-  "port-version": 2,
+  "port-version": 3,
   "description": "A fast JSON parser/generator for C++ with both SAX/DOM style API <http://rapidjson.org/>",
   "homepage": "http://rapidjson.org/",
   "license": "MIT",
diff --git a/versions/baseline.json b/versions/baseline.json
index 1d9b94fb0..8a2a496ac 100644
--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -6194,7 +6194,7 @@
     },
     "rapidjson": {
       "baseline": "2022-06-28",
-      "port-version": 2
+      "port-version": 3
     },
     "rapidxml": {
       "baseline": "1.13",
diff --git a/versions/r-/rapidjson.json b/versions/r-/rapidjson.json
index 37386b453..9c68fb21c 100644
--- a/versions/r-/rapidjson.json
+++ b/versions/r-/rapidjson.json
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "42979842f5b3e16d7b502c32f9fe2073add65ab2",
+      "version-date": "2022-06-28",
+      "port-version": 3
+    },
     {
       "git-tree": "f3770e13ddf21ddf223c84dab78f3d04d4f25750",
       "version-date": "2022-06-28",
