diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6f83c07..3750147 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,13 +39,9 @@ include_directories(${OpenCV_INCLUDE_DIRS})
 find_package(Boost REQUIRED)
 include_directories(${Boost_INCLUDE_DIR})
 
-find_package(DLib QUIET 
-  PATHS ${DEPENDENCY_INSTALL_DIR})
-if(${DLib_FOUND})
-  message("DLib library found, using it from the system")
-  include_directories(${DLib_INCLUDE_DIRS})
-  add_custom_target(Dependencies)
-else(${DLib_FOUND})
+find_package(dlib CONFIG REQUIRED)
+include_directories(${Dlib_INCLUDE_DIR})
+if(0)
   message("DLib library not found in the system, it will be downloaded on build")
   option(DOWNLOAD_DLib_dependency "Download DLib dependency" ON)
   if(${DOWNLOAD_DLib_dependency})
@@ -59,13 +55,13 @@ else(${DLib_FOUND})
   else()
     message(SEND_ERROR "Please, activate DOWNLOAD_DLib_dependency option or download manually")
   endif(${DOWNLOAD_DLib_dependency})
-endif(${DLib_FOUND})
+endif()
 
 if(BUILD_DBoW2)
   add_library(${PROJECT_NAME} SHARED ${SRCS})
   include_directories(include/DBoW2/)
   add_dependencies(${PROJECT_NAME} Dependencies)
-  target_link_libraries(${PROJECT_NAME} ${OpenCV_LIBS} ${DLib_LIBS})
+  target_link_libraries(${PROJECT_NAME} ${OpenCV_LIBS} dlib::dlib)
 endif(BUILD_DBoW2)
 
 if(BUILD_Demo)
